amy/recruitment/views.py

```diff
@@ -340,6 +341,16 @@ def form_valid(self, form: InstructorRecruitmentAddSignupForm) -> HttpResponse:
         signup: InstructorRecruitmentSignup = form.save(commit=False)
         signup.recruitment = self.object
         signup.save()
+
+        admin_signs_instructor_up_for_workshop_signal.send(
+            sender=signup,
+            request=self.request,
+            person_id=signup.person.pk,
+            event_id=signup.recruitment.event.pk,
+            instructor_recruitment_id=signup.recruitment.pk,
+            instructor_recruitment_signup_id=signup.pk,
```

Comment:
Design question: Since most of this information is coming from the `sender`, why do we send all these args separately?

Reply:
Good question. So the sender theoretically can change: https://docs.djangoproject.com/en/4.2/topics/signals/#connecting-to-signals-sent-by-specific-senders

We can have different senders sending the same signal. A good example is `Instructor Signs Up For Workshop`. It could be sent by instructor itself, but it could be also sent by admin when it's the admin who signs instructor up for a workshop on their behalf.

We actually split this case into two separate signals (`instructor_signs_up_for_workshop` and `admin_signs_instructor_up_for_workshop`), but Django's design doesn't limit us in the value of `sender`, so this could potentially be anything... And as far as I remember, it's event typed like so: `sender: Any`.

Given this lack of constraint on the sender I decided to use typing system to explicitly specify what arguments are required for each and every signal we have. Thus the use of `Unpack` on `kwargs`.